spring-rest-http/src/test/java/com/baeldung/controller/ExceptionControllerUnitTest.java

```diff
@@ -0,0 +1,65 @@
+package com.baeldung.controller;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.baeldung.controllers.ExceptionController;
+import com.baeldung.exception.BadArgumentsException;
+import com.baeldung.exception.InternalException;
+import com.baeldung.exception.ResourceNotFoundException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ExceptionController.class)
+public class ExceptionControllerUnitTest{
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Test
+    public void givenNotFound_whenGetSpecificException_thenNotFoundCode() throws Exception {
+        String exceptionParam = "not_found";
+
+        mvc.perform(get("/exception/{exception_id}", exceptionParam)
+            .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isNotFound())
+            .andExpect(result -> assertTrue(result.getResolvedException() instanceof ResourceNotFoundException))
+            .andExpect(result -> assertEquals("resource not found", result.getResolvedException().getMessage()));

```

Comment:
doesn't sound a good idea to assert messages. maybe we could go for an error code solution?

Reply:
I'm already asserting the response code, so what I would like to show with the message assertion is that it is also possible to check if the exception contains the expected message. I think that it would be useful for the reader.

Reply:
yes but I was suggesting we create that instead of asserting or even sending the exceptions, we send a code (not an HTTP code) stating the failure. It could even be an enumeration, like "RESOURCE_NOT_FOUND". But asserting exception messages sounds wrong and we should kind of suggest to our readers that it's ok.

Reply:
If I understand correctly, what we should do here is to replace the plain String to a constant or an enumeration, Is it correct?